tests/unit/demography/test_t_model_functions.py

```diff
@@ -1,21 +1,13 @@
 """test the functions in t_model_functions.py."""
 
 import numpy as np
-from numpy.testing import assert_array_almost_equal
-
-from pyrealm.demography.t_model_functions import (
-    calculate_canopy_q_m,
-    calculate_crown_areas,
-    calculate_crown_fractions,
-    calculate_foliage_masses,
-    calculate_heights,
-    calculate_sapwood_masses,
-    calculate_stem_masses,
-)
 
 
 def test_calculate_heights():
     """Tests happy path for calculation of heights of tree from diameter."""
+
+    from pyrealm.demography.t_model_functions import calculate_heights
```

Comment:
Hm, I don't know how I feel about moving imports into functions. I prefer them to be on the top of the file, for better overview. Is there a difference in performance if you draw the imports in here?

Reply:
(same below)

Reply:
That style was advice from the RSE team at Imperial. The argument is that having the imports for the **package being tested** inside the functions isolates the actual importatation as part of the test. So, if stuff goes wrong with an import, a specific related test fails, rather than a whole module of tests (or the whole test suite).

I definitely see what you mean about having an overview of which code is getting tested but I buy their argument and we've seen it be useful in practice.